contracts/lib/LibPrices.sol

```diff
@@ -96,48 +134,45 @@ library Prices {
         PriceInstant[24] memory tracerPrices,
         PriceInstant[24] memory oraclePrices
     ) public pure returns (TWAP memory) {
-        uint256 instantDerivative = 0;
+        uint256 totalDerivativeTimeWeight = 0;
+        uint256 totalUnderlyingTimeWeight = 0;
         uint256 cumulativeDerivative = 0;
-        uint256 instantUnderlying = 0;
         uint256 cumulativeUnderlying = 0;
 
         for (uint256 i = 0; i < 8; i++) {
             uint256 currTimeWeight = 8 - i;
             // if hour < i loop back towards 0 from 23.
             // otherwise move from hour towards 0
-            uint256 j = hour < i ? 23 - i + hour : hour - i;
+            uint256 j = hour < i ? 24 - i + hour : hour - i;
 
             uint256 currDerivativePrice = averagePrice(tracerPrices[j]);
```

Comment:
If `hour` is greater than or equal to 24, these two array accesses will be out-of-bounds, correct?

Reply:
Yeah, how should this behave if `hour >= 24`? 
@kumar-ish

Reply:
It shouldn't be valid; I guess we can do an explicit check/require for it, or otherwise it'll just revert

Reply:
Don't believe we can do a require because library functions should be pure but yep agreed

Reply:
Added `require` for hour check